Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -312,19 +312,61 @@ private func expandAccessorMacroWithExistingAccessors(
   return "\(raw: indentedSource)"
 }
 
+private func buildQualifiedType(_ inner: DeclSyntax) -> TypeSyntax? {
+  if let ext = inner.as(ExtensionDeclSyntax.self) {
+    return ext.extendedType.trimmed
+  }
+
+  guard let identified = inner.asProtocol(NamedDeclSyntax.self) else {
+    return nil
+  }
+
+  var typeNames: [TokenSyntax] = []
+  var parentExtension: TypeSyntax? = nil
+  var possibleParent = inner.parent
+  while let parent = possibleParent {
+    possibleParent = parent.parent
+    if let parent = parent.asProtocol(NamedDeclSyntax.self) {
+      typeNames.append(parent.name.trimmed)
+    } else if let parent = parent.as(ExtensionDeclSyntax.self) {
+      // Extensions must be top level
+      if parentExtension != nil {
+        return nil
+      }
+      parentExtension = parent.extendedType.trimmed
+    }
+  }
+
+  // Attached type is within an extension
+  if let parentExtension {
+    // Extensions must be top level
+    if !typeNames.isEmpty {
+      return nil
+    }
+    return TypeSyntax(MemberTypeSyntax(baseType: parentExtension, name: identified.name.trimmed))
```

Comment:
We can't discard `AnotherWrapper` in this case.
```swift
      struct Wrapper {}
      extension Wrapper {
        struct AnotherWrapper {
          @AddSendableExtension
          struct MyType {}
        }
      }
```

Reply:
Something like this might be useful.
```swift
extension SyntaxProtocol {
  /// Get qualified type name.
  /// For example, for `struct Foo { struct Bar {} }`,
  /// calling on the inner struct returns `Foo.Bar`.
  var syntacticQualifiedTypeContext: TypeSyntax? {
    let type = self.parent?.syntacticQualifiedTypeContext

    // Don't handle nested 'extension' cases. They are invalid anyway.
    if type == nil, let ext = self.as(ExtensionDeclSyntax.self) {
      return ext.extendedType
    } else if let named = self.asProtocol(NamedDeclSyntax.self) {
      if let base = type {
        return TypeSyntax(MemberTypeSyntax(baseType: base, name: named.name.trimmed))
      } else {
        return TypeSyntax(IdentifierTypeSyntax(name: named.name.trimmed))
      }
    }
    return type
  }
}
```